### PR TITLE
rebuilding to React

### PR DESCRIPTION
**Pull request by Saturday17**

### CODE REVIEW

File: react-app/src/components/Arrowdown.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, {Component} from 'react';
+
+class Arrowdown extends React.Component {
+
+  render(){
+    return (
+      <>
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Arrowdown.js]: If you return only one JSX element then you don't need wrap this element in Fragment

File: react-app/src/components/Arrowdown.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, {Component} from 'react';
+
+class Arrowdown extends React.Component {
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Arrowdown.js]: If you import `{Component}` then extends your custom component from `Component`, example
`class Arrowdown extends Component {`

File: react-app/src/components/Authorization.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react';
+
+class Authorization extends React.Component {
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Authorization.js]: If you import `{Component}` then extends your custom component from `Component`, example
`class Arrowdown extends Component {`

File: react-app/src/components/Arrowdown.js (hunk 1 above)
Comment (Saturday17) [reply react-app/src/components/Arrowdown.js]: Yes, I did`t pay attetion to this

File: react-app/src/components/Authorization.js (hunk 3 above)
Comment (Saturday17) [reply react-app/src/components/Authorization.js]: Ok, thanks

File: react-app/src/components/Authorization.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, {Component} from 'react';
+
+class Authorization extends Component {
+  openRegistration(event){
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Authorization.js:4]: Don't use `.addEventListener`. We have parent components, which could handle such clicks via props callbacks. Use state variable and make method `handleTriggerRegistrationModal` or something like that and pass this method via props in `<Authorization onTriggerModal={this.handleTriggerRegistrationModal}>` component

File: react-app/src/components/Authorization.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          </div>
+          <div className="form-field input-right">
+            <label htmlFor="pass">Password</label>
+            <input type="password" name="pass" id="pass" placeholder="your password"/>
+          </div>
+          <div className="submit-field input-right" value="Регистрация">
+            <button>Sign in</button>
+          </div>
+          <div className="reg-link reg-btn">
+            <a href="#" onClick={ this.openRegistration }>Registration</a>
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Authorization.js:33]: And here you can use props callback:
`<a href="#" onClick={ this.props.onTriggerModal }>Registration</a>`

File: react-app/src/components/Authorization.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    formAuthorization.removeAttribute('opened');
+    formRegistration.addEventListener('click', function(e){
+      if(this === e.target){
+          formRegistration.removeAttribute('opened');
+      }
+    });
+  }
+  render(){
+    return (
+      <div className="form-container" id="form-authorization">
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Authorization.js:18]: Do you know BEM naming? Links:
https://medium.freecodecamp.org/a-quick-introduction-to-block-element-modifiers-bem-9df46d29b64c
https://css-tricks.com/bem-101/
https://www.smashingmagazine.com/2018/06/bem-for-beginners/

Please use BEM

File: react-app/src/components/Calltoaction.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import React, {Component} from 'react';
+
+class Calltoaction extends Component {
+
+  openAuthorization(event){
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Calltoaction.js:5]: same as before, use callback from parent component

File: react-app/src/components/Calltoaction.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    let formRegistration = document.getElementById('form-registration');
+    formAuthorization.setAttribute('opened', 'true');
+    formRegistration.removeAttribute('opened');
+    formAuthorization.addEventListener('click', function(e){
+        if(this === e.target){
+            formAuthorization.removeAttribute('opened');
+        }
+    });
+  }
+  openRegistration(event){
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Calltoaction.js:17]: same as before, user callback from props

File: react-app/src/components/Foot.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import React, {Component} from 'react';
+import logo from '../img/logomin.png';
+
+class Foot extends Component {
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Foot.js:4]: Rename to the Footer, Foot is not clear. What does Foot mean? Leg? :)

File: react-app/src/components/Foot.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    });
+  }
+  render(){
+    return (
+      <>
+        <div className="footer">
+          <div className="logo">
+            <a href="#"> <img src={logo} alt="logo"/> </a>
+          </div>
+          <div className="footer-menu">
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Foot.js:24]: You could move it in independent component.

File: react-app/src/components/Foot.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+            <ul>
+              <li><a href="#" className="footer-menu-link">Affiche</a></li>
+              <li><a href="#" className="footer-menu-link">Cinemas</a></li>
+              <li><a href="#" className="footer-menu-link">Food and Drinks</a></li>
+              <li><a href="#" className="footer-menu-link">Info</a></li>
+              <li><a href="#" className="footer-sign-in sign-in-btn footer-menu-link" onClick={ this.openAuthorization }>Sign in</a></li>
+            </ul>
+          </div>
+        </div>
+        <div className="line"></div>
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Foot.js:34]: line? why do we need empty tag?

File: react-app/src/components/Menu.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    formAuthorization.addEventListener('click', function(e){
+        if(this === e.target){
+            formAuthorization.removeAttribute('opened');
+        }
+    });
+  }
+  render(){
+    return (
+      <>
+        <Minimenu />
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Menu.js:38]: camelCase

File: react-app/src/components/Menu.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+              <i className="fa fa-bars"></i>	
+          </a>
+          <div className="menu visible-lg">
+              <ul><li><a href="#" className="menu-link">Affiche</a></li>
+              <li><a href="#" className="menu-link">Cinemas</a></li>
+              <li><a href="#" className="menu-link">Food and Drinks</a></li>
+              <li><a href="#" className="menu-link">Info</a></li>
+              <li>
+                  <form className="form-search">
+                      <input type="text" placeholder="Поиск по сайту" className="input-search" />
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Menu.js:53]: don't use russian

File: react-app/src/components/Menu.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+import Minimenu from './Minimenu'
+
+class Menu extends Component {
+
+  openMmenu(event){
+    let menu = document.getElementById('m-menu');
+    event.preventDefault();
+    menu.classList.toggle('m-menu__active');
+  }
+  searchInput(event){
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Menu.js:12]: use css for it,  <selector>:focus

File: react-app/src/components/Minimenu.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
+import React, {Component} from 'react';
+
+class Minimenu extends Component {
+  closeMmenu(event){
+    let menu = document.getElementById('m-menu');
+    event.preventDefault();
+    menu.classList.toggle('m-menu__active');
+  }
+  searchInput(event){
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Minimenu.js:9]: use CSS

File: react-app/src/components/Minimenu.js [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+    formAuthorization.addEventListener('click', function(e){
+        if(this === e.target){
+            formAuthorization.removeAttribute('opened');
+        }
+    });
+  } 
+  render(){
+    return (
+      <ul className="m-menu" id="m-menu">
+        <li className="close-menu" onClick={ this.closeMmenu }>&times;</li>
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Minimenu.js:35]: Mmenu?

File: react-app/src/components/Minimenu.js [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+        <li><a href="#" className="menu-link">Affiche</a></li>
+        <li><a href="#" className="menu-link">Cinemas</a></li>
+        <li><a href="#" className="menu-link">Food and Drinks</a></li>
+        <li><a href="#" className="menu-link">Info</a></li>
+        <li>
+            <form className="form-search">
+                <input type="text" placeholder="Поиск по сайту" className="input-search"/>
+            </form>
+        </li>
+        <li><i className="fa fa-search search-icon" style={{color: 'rgba(255, 255, 255, 1)'}} onClick={ this.searchInput }></i></li>
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Minimenu.js:45]: don't use inline styling

File: react-app/src/components/Minimenu.js [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    return (
+      <ul className="m-menu" id="m-menu">
+        <li className="close-menu" onClick={ this.closeMmenu }>&times;</li>
+        <li><a href="#" className="menu-link">Affiche</a></li>
+        <li><a href="#" className="menu-link">Cinemas</a></li>
+        <li><a href="#" className="menu-link">Food and Drinks</a></li>
+        <li><a href="#" className="menu-link">Info</a></li>
+        <li>
+            <form className="form-search">
+                <input type="text" placeholder="Поиск по сайту" className="input-search"/>
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Minimenu.js:42]: russian, english only

File: react-app/src/components/Minimenu.js [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+    let formSearches = document.querySelectorAll('.input-search');
+    [].forEach.call(formSearches, function(item){
+        if(item.className == 'input-search'){
+            item.classList.add('input-search__active');
+        } else if(item.className == 'input-search input-search__active'){
+            item.classList.remove('input-search__active');
+        }
+    });
+  }
+  openAuthorization(event){
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Minimenu.js:20]: as before, define method in parent component and pass it via props

File: react-app/src/components/Table.js [hunk 20]
```diff
@@ -0,0 +1,39 @@
+import React, {Component} from 'react';
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Table.js:1]: please move information about films in some json/js objects and reuse such data from these objects. Let's imitate real data from backend.

File: react-app/src/components/Table.js [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+class Table extends Component {
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Table.js:3]: if your component doesn't have state or lifecycle methods, then you should use functional component.

File: react-app/src/components/Textinfo.js [hunk 22]
```diff
@@ -0,0 +1,18 @@
+import React, {Component} from 'react';
+
+class Textinfo extends Component {
```
Comment (siarhei-arzamasau) [inline react-app/src/components/Textinfo.js:3]: functional component

File: react-app/src/index.js [hunk 23]
```diff
@@ -2,92 +2,46 @@ import React, { Component } from 'react';
...
-        event.preventDefault();
-        let formSearches = document.querySelectorAll('.input-search');
-        [].forEach.call(formSearches, function(item){
-            if(item.opened) {
-                item.removeAttribute('opened');
-            } else if(!item.opened) {
-                item.setAttribute('opened', 'true');
+class Index extends React.Component {
+    componentDidMount(){
+        window.addEventListener('keydown', function(e){
```
Comment (siarhei-arzamasau) [inline react-app/src/index.js:17]: use onKeyDown from React

File: react-app/src/components/Authorization.js (hunk 4 above)
Comment (Saturday17) [reply react-app/src/components/Authorization.js:4]: Ok, I need to get rid of old habits)

File: react-app/src/components/Authorization.js (hunk 6 above)
Comment (Saturday17) [reply react-app/src/components/Authorization.js:18]: Ok, I need to familiarize with BEM

File: react-app/src/components/Foot.js (hunk 9 above)
Comment (Saturday17) [reply react-app/src/components/Foot.js:4]: Yes, sounds funny)

File: react-app/src/components/Foot.js (hunk 11 above)
Comment (Saturday17) [reply react-app/src/components/Foot.js:34]: I delete this useless tag, and made hr tag

File: react-app/src/components/Foot.js (hunk 10 above)
Comment (Saturday17) [reply react-app/src/components/Foot.js:24]: ok

File: react-app/src/components/Menu.js (hunk 12 above)
Comment (Saturday17) [reply react-app/src/components/Menu.js:38]: Names of components too? But they must named with upper case as i know, => MiniMenu ?

File: react-app/src/components/Foot.js (hunk 11 above)
Comment (siarhei-arzamasau) [reply react-app/src/components/Foot.js:34]: In some cases you can use `border-bottom`

File: react-app/src/components/Menu.js (hunk 12 above)
Comment (siarhei-arzamasau) [reply react-app/src/components/Menu.js:38]: yes, `MiniMenu`, other example: `MyFavouriteDropDown`

**Pull request merged**